Add setup instructions (#7)
mkcor/python-prog: README.md

lessons, which are available under the
[CC-BY license](https://creativecommons.org/licenses/by/4.0/).

> Please refer to the [Setup](#setup) section to install the required software before the start of the tutorial.

Reviewer: ```suggestion
Please refer to the [Setup](#setup) section to install the required software ahead of the tutorial.
```

## Setup

> ### Installing Python Using Anaconda

Reviewer 2: I would not think that the beginners would be able to judge if to install Anaconda or Miniconda and why they are given two choices. I would suggest just leaving the miniconda instructions

Reviewer: I agree with @maikia. I would keep a single set of instructions, namely those using Miniconda.

2. Click on the latest Installer link depending on your OS: **Miniconda3 Windows 64-bit** for Windows, **Miniconda3 macOS Intel x86 64-bit pkg** for MacOS, **Miniconda3 Linux 64-bit** for Linux.

> 3. Install Python 3 by running the Anaconda Installer (double click on the downloaded file) using all of the defaults for installation. On MacOS and Linux, make sure to check **Add Miniconda to my PATH environment variable**.

Reviewer: ```suggestion
3. Install Python 3 by running the Miniconda Installer (double click on the downloaded file) using all of the defaults for installation. On MacOS and Linux, make sure to check **Add Miniconda to my PATH environment variable**.
```
?

>    

Reviewer: ```suggestion
```
(for consistency)

> 4. On Windows: open the **Anaconda Prompt** from the **Start** menu. On MacOs and Linux: open the **Terminal** app. Install the required packages by running the following lines in the opened window:

Reviewer: ```suggestion
4. On Windows: open the **Anaconda Prompt** from the **Start** menu. On MacOs and Linux: open the **Terminal** app. Install the required packages by running the following lines (copy-paste them in the window that just opened up and press Enter):
```
(being super explicit, as discussed with @maikia)

```
conda config --add channels conda-forge

> conda install jupyterlab=3.5.3 matplotlib=3.7.0 ipywidgets=7.6.5 voila=0.3.6 pandas=1.5.3

Reviewer: No big deal, but could we have the packages in alphabetical order?

```

> 5. Type `jupyter lab` in the Anaconda Prompt / Terminal. After Jupyter Lab has launched, click the “Python 3” button under “Notebook” in the launcher window, or use the “File” menu, to open a new Python 3 notebook.

Reviewer: ```suggestion
5. Type `jupyter lab` in the Anaconda Prompt / Terminal. After JupyterLab has launched, click the “Python 3” button under “Notebook” in the launcher window, or use the “File” menu, to open a new Python 3 notebook.
```

> 6. To test your environment, run the following lines in a cell of the notebook:

Reviewer: ```suggestion
6. To test your setup, run the following lines in a cell of the notebook:
```

```
import pandas as pd                                                            
                                                                               
table = pd.DataFrame(
    {'Time': [0, 1, 2, 3],
     'Emma': [0, 10, 20, 30]}
)  
table.plot();
```

> 
> 

Reviewer: ```suggestion
```

> 

Reviewer 2: This is great. I would still suggest to add some instructions to check if everything is installed correctly (ie, just like you do with Python but for Jupyter and for importing the libraries within Jupyter).

Reviewer 2: Maybe also would be worth adding a link to Software Carpentry installation instructions for troubleshooting?

Reviewer: @maikia hmm, the Software Carpentry setup instructions I know of are for [Installing Python Using Anaconda](http://swcarpentry.github.io/python-novice-gapminder/setup.html#installing-python-using-anaconda)... So, pointing to those would contradict our previous review [comment](https://github.com/mkcor/python-prog/pull/7/files#r1156397840).

Instead, I was going to suggest adding a subsection for testing the setup (e.g., see last point of https://datacarpentry.org/image-processing/setup/). The test could consist in launching JupyterLab and running a code cell (I know that students will be learning how to do so in the first episode of this tutorial, but, well) such as:

```py
import pandas as pd                                                            
                                                                               
table = pd.DataFrame(
    {'Time': [0, 1, 2, 3],
     'Emma': [0, 10, 20, 30]}
)  
table.plot();
```
They would copy-paste the above code snippet and check that a plot shows up right below the code cell.

@bmaz does that make sense?
